### PR TITLE
Adapt freeipa.spec.in for latest Fedora, fix python2 ipatests packaging bug

### PR DESCRIPTION
**Pull request by t-woerner**
New autoreconf -ivf call before configure

### CODE REVIEW

File: freeipa.spec.in [hunk 1]
```diff
@@ -35,6 +37,9 @@
 %if 0%{?fedora} >= 29
     # F29 only supports Python 3 as default Python
     %global with_default_python 3
+    %if 0%{?fedora} >= 30
+        %global with_python2 0
```
Comment (tiran) [inline freeipa.spec.in]: This change hasn't been accepted yet. We need to provide Python 2 packages until Fedora has accepted removal of Python 2 packages.

File: freeipa.spec.in [hunk 2]
```diff
@@ -961,6 +967,8 @@ find \
 	-type f -exec grep -qsm1 '^#!.*\bpython' {} \; \
 	-exec sed -i -e '1 s|^#!.*\bpython[^ ]*|#!%{__python2}|' {} \;
 
+aclocal --force
```
Comment (tiran) [inline freeipa.spec.in]: Why is this necessary? The tar balls should contain configure script.

Comment (t-woerner) [reply freeipa.spec.in]: The build issues that we had on F-29 are because of not having aclocal v1.15. There is only aclocal v1.16. If you are adding patches, then it is needed to run aclocal and autoconf with the force option.

Comment (abbra) [reply freeipa.spec.in]: @tiran we've been hit twice already in F28 lifetime when autotools were upgraded in Fedora, first to 1.15 and now to 1.16. Actual autotools version is hardcoded in the generated script so you need to have exact environment to the one where tarball was generated or you have to rebuild it.

Comment (abbra) [reply freeipa.spec.in]: We also have to re-run aclocal/autoconf for any change with Makefile.am, so let us just re-run it always.

Comment (rcritten) [reply freeipa.spec.in]: Why not use autoreconf -ivf to ensure everything is up to date?

**Comment by t-woerner** [comment]
Backporting for 4.7.0 might be good as the current spec file has an issue with python2 builds.

**Comment by tiran** [comment]
master:

* 1d3c308bf174897b38f076f0d3c036948989204c Adapt freeipa.spec.in for latest Fedora, fix python2 ipatests packaging bug
